File: 14LongestCommonPrefix/longestCommonPrefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    let prefix = strs[0]
    let prefixLength = prefix.length

    for (let i = 1; i < strs.length; i++) {
        let currentStrs = strs[i]
        while (prefix !== currentStrs.substring(0, prefixLength)) {
            prefixLength--
            if (prefixLength === 0) {
                return ''
            }
            prefix = prefix.substring(0, prefixLength)
        }
    }
    return prefix
};

//I'm wondering what "easy" means in leetcode, this actually feels fairly complicated

console.log(longestCommonPrefix(["flower", "flow", "flight"]))

File: 14LongestCommonPrefix/longestCommonPrefix.cpp
#include <iostream>
#include <vector> 

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        if(strs.size() == 0){
            return "";
        }

        string prefix = strs[0];
        for (int i = 0; i != strs.size(); ++i){
            while(strs[i].find(prefix) != 0){
                prefix = prefix.substr(0, prefix.size() - 1);
                //cout << prefix << endl;
            }
        }
        return prefix;
    }
};

int main(){
        Solution sol;
        vector<string> strings = {"flower", "flow", "flight"};
        cout << sol.longestCommonPrefix(strings) << endl;
        return 0;
    };